src/base/BaseSystem.spec.ts

```diff
@@ -1,33 +1,37 @@
+import { createTestScene } from "../testutils/SceneMock";
 import { BaseSystem } from "./BaseSystem";
 import { ComponentType } from "./Component";
-import { createTestScene } from "./Game.test";
 import { Scene } from "./Scene";
 
-describe(module.id, () => {
-    const TestComponent1 = new ComponentType<"testComponent1", "1">("testComponent1");
-    const TestComponent2 = new ComponentType<"testComponent2", "2">("testComponent2");
-
-    class TestBaseSystem extends BaseSystem<[typeof TestComponent1, typeof TestComponent2]> {
-        constructor(scene: Scene) {
-            super(scene, [TestComponent1, TestComponent2], []);
-        }
+const TestComponent1 = new ComponentType<"testComponent1", "1">(
+    "testComponent1"
+);
+const TestComponent2 = new ComponentType<"testComponent2", "2">(
+    "testComponent2"
+);
+class TestBaseSystem extends BaseSystem<
+    [typeof TestComponent1, typeof TestComponent2]
+> {
+    constructor(scene: Scene) {
+        super(scene, [TestComponent1, TestComponent2], []);
     }
+}
 
-    it("should only track entities with all required components", () => {
-        const {scene} = createTestScene([TestComponent1, TestComponent2]);
-
-        const system = scene.addSystem(TestBaseSystem);
+describe(module.id, () => {
+    let scene: Scene;
+    let system: TestBaseSystem;
 
-        expect(system.getTrackedCount()).toBe(0);
+    beforeEach(() => {
+        scene = createTestScene([TestComponent1, TestComponent2]).scene;
```

Comment:
Generally I prefer a setup function with `const` returns over mutable `describe`-level `let`s since you can get things in a weird state with `let`s and `const`s are easier to compare.

Reply:
Something like this, that we can just decompose:
```ts
const setup = () => {
    const scene = createTestScene([TestComponent1, TestComponent2]).scene;
    const system = scene.addSystem(TestBaseSystem);
    return {scene, system};
}
```